Objects/typeobject.c

```diff
@@ -1761,6 +1761,36 @@ mro_implementation(PyTypeObject *type)
             return NULL;
     }
 
+    bases = type->tp_bases;
+    n = PyTuple_GET_SIZE(bases);
+    if (n == 1) {
+        /* Fast path: if there is a single base, constructing the MRO
+         * is trivial.
+         */
+        PyTypeObject *base = (PyTypeObject *)PyTuple_GET_ITEM(bases, 0);
+        Py_ssize_t k;
+
+        if (base->tp_mro == NULL) {
+            PyErr_Format(PyExc_TypeError,
+                         "Cannot extend an incomplete type '%.100s'",
+                         base->tp_name);
+            return NULL;
+        }
+        k = PyTuple_GET_SIZE(base->tp_mro);
+        result = PyTuple_New(k + 1);
```

Comment:
Shouldn't the result be a tuple?

Reply:
You mean a list? The caller calls `PySequence_Tuple` on the result, so it's faster if we create a tuple upfront.